cmd/agent-sandbox-controller/main.go

```diff
@@ -56,13 +67,42 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	// Importing net/http/pprof registers handlers on the global DefaultServeMux.
+	// Reset it to avoid accidentally exposing pprof via any server that uses the default mux.
+	http.DefaultServeMux = http.NewServeMux()
+
 	scheme := controllers.Scheme
 	if extensions {
 		utilruntime.Must(extensionsv1alpha1.AddToScheme(scheme))
 	}
 
+	metricsOpts := metricsserver.Options{
+		BindAddress: metricsAddr,
+	}
+	if enablePprof || enablePprofDebug {
+		setupLog.Info("pprof enabled", "debug", enablePprofDebug)
+		metricsOpts.ExtraHandlers = map[string]http.Handler{
+			"/debug/pprof/profile": http.HandlerFunc(pprof.Profile),
+		}
+		if enablePprofDebug {
+			setupLog.Info("pprof debug endpoints enabled", "debug", enablePprofDebug)
+			runtime.SetBlockProfileRate(1)     // 1 = sample all events
+			runtime.SetMutexProfileFraction(1) // 1 = sample all events
```

Comment:
Similarly, `runtime.SetMutexProfileFraction(1)` samples every mutex contention event. This is very expensive. A value like `10` or `100` (sampling 10% or 1%) is usually sufficient for debugging contention without killing performance.